fili-core/src/main/java/com/yahoo/bard/webservice/web/ratelimit/DefaultRateLimiter.java

```diff
@@ -0,0 +1,250 @@
+
+        // Register meters for number of requests
+        requestUserMeter = REGISTRY.meter("ratelimit.meter.request.user");
+        requestUiMeter = REGISTRY.meter("ratelimit.meter.request.ui");
+        requestBypassMeter = REGISTRY.meter("ratelimit.meter.request.bypass");
+        rejectUserMeter = REGISTRY.meter("ratelimit.meter.reject.user");
+        rejectUiMeter = REGISTRY.meter("ratelimit.meter.reject.ui");
+    }
+
+    /**
+     * Get the current count for this username. If user does not have a counter, create one.
+     *
+     * @param userName  Username to get the count for
+     *
+     * @return The atomic count for the user
+     */
+    private AtomicInteger getCount(String userName) {
+        AtomicInteger count = userCounts.get(userName);
+
+        // Create a counter if we don't have one yet
+        if (count == null) {
+            userCounts.putIfAbsent(userName, new AtomicInteger());
+            count = userCounts.get(userName);
+            usersCounter.inc();
+        }
+
+        return count;
+    }
+
+    /**
+     * Increment the initial count and check if the count has gone over the request limit.
+     *
+     * @param initialCount  Initial count that we're incrementing and checking against the limit
+     * @param requestLimit  Limit to check the incremented initial count against
+     *
+     * @return True if the incremented count is <= the request limit, false if it's gone over and the request limit
+     * isn't the DISABLED_RATE (-1).
+     */
+    private boolean incrementAndCheckCount(AtomicInteger initialCount, int requestLimit) {
+        int count = initialCount.incrementAndGet();
+        if (count > requestLimit && requestLimit != DISABLED_RATE) {
+            initialCount.decrementAndGet();
+            LOG.info("reject: {} > {}", count, requestLimit);
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Do the house keeping needed to reject the request.
+     *
+     * @param rejectMeter  Meter to count the rejection in
+     * @param isRejectGlobal  Whether or not the rejection is on the global rate limit
+     * @param isUIQuery  Whether or not the request is a UI Query
+     * @param userName  Username of the user who made the request
+     */
+    private void rejectRequest(Meter rejectMeter, boolean isRejectGlobal, boolean isUIQuery, String userName) {
+        rejectMeter.mark();
+        String limitType = isRejectGlobal ? "GLOBAL" : isUIQuery ? "UI" : "USER";
+        LOG.info("{} limit {}", limitType, userName);
+    }
+
+    @Override
+    public RateLimitRequestToken getToken(ContainerRequestContext request) {
+        MultivaluedMap<String, String> headers = Utils.headersToLowerCase(request.getHeaders());
+
+        if (DataApiRequestTypeIdentifier.isBypass(headers) ||
+                DataApiRequestTypeIdentifier.isCorsPreflight(request.getMethod(), request.getSecurityContext())) {
```

Comment:
Close and open parenthesis need to balance on indentation.

Reply:
indent the first part if wrapping.

Reply:
done